base_search_mail_content/models/mail_thread.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# © 2016-17 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# © 2016 Serpent Consulting Services Pvt. Ltd. (<http://www.serpentcs.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from lxml import etree
+from odoo.osv import expression
+from odoo.osv.orm import setup_modifiers
+
+
+class MailThread(models.AbstractModel):
+
+    _inherit = 'mail.thread'
+
+    def _search_message_content(self, operator, value):
+
+        main_operator = 'in'
+        if operator in expression.NEGATIVE_TERM_OPERATORS:
+            main_operator = 'not in'
+            operators = {'!=': '=', 'not like': 'like',
+                         'not ilike': 'ilike', 'not in': 'in'}
+            operator = operators[operator]
+        domain = [('model', '=', self._name), '|', '|', '|', '|',
+                  ('record_name', operator, value),
+                  ('subject', operator, value), ('body', operator, value),
+                  ('email_from', operator, value),
+                  ('reply_to', operator, value)]
+        recs = self.env['mail.message'].search(domain)
+        return [('id', main_operator, recs.mapped('res_id'))]
+
+    @api.multi
+    def _compute_message_content(self):
+        """ We don't really need to show any content. This field is to be
+        used only by searches"""
+        pass
+
+    message_content = fields.Text(
+        string='Message Content',
+        help='Message content, to be used only in searches',
+        compute="_compute_message_content",
+        search='_search_message_content')
+
+
+class BaseModel(object):
+
+    _inherit = 'base'
```

Comment:
Yeah, I know 😆 

Sorry for confusing. I said that before because it's the new way to avoid monkey patches, but after seeing this now, I realized that you only need this code for addons that inherit from `mail.thread`, so possibly adding the method there should be better.